src/Propel/Runtime/ActiveQuery/Criteria.php

```diff
@@ -1247,6 +1255,41 @@ public function hasSelectModifier($modifier)
         return in_array($modifier, $this->selectModifiers);
     }
 
+    /**
+     * @return null|Lock Get lock value.
+     */
+    public function getLock()
+    {
+        return $this->lock;
+    }
+
+    /**
+     * Apply a read lock to be used to retrieve rows.
+     *
+     * @see self::SHARED
+     * @see self::EXCLUSIVE
+     *
+     * @return $this Modified Criteria object (for fluent API)
+     */
+    public function withLock(string $lockType, array $tableNames = [], $noWait = false)
```

Comment:
The param docblocks seem missing for these.
Also, could $noWait be bool typehinted?

Reply:
Sure, I could convert all my code to be properly typehinted according to minimum php reqs which is 7.2

Reply:
done